Unable to upgrade to graylog 3.3

<!--- Provide a general summary of the issue in the Title above -->
Graylog fails to start upfer upgrade to version 3.3 to 3.2

## Expected Behavior
Graylog starting to work and become available
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
Fails with error log 
```
2020-09-22T11:25:34.551+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.551+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.551+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.551+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.551+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.552+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.552+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.552+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.552+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
2020-09-22T11:25:34.552+02:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Error injecting constructor, java.lang.NullPointerException
```
Reading the messages log from the server I can find these messages:
```
Sep 22 11:25:34 hostname graylog-server: Exception in thread "main" com.google.inject.CreationException: Unable to create injector, see the following errors:
Sep 22 11:25:34 hostname graylog-server: 1) Error injecting constructor, java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.buffers.ProcessBuffer.<init>(ProcessBuffer.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:59)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.buffers.ProcessBuffer
Sep 22 11:25:34 hostname graylog-server: for the 2nd parameter of org.graylog2.periodical.ThrottleStateUpdaterThread.<init>(ThrottleStateUpdaterThread.java:76)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.periodical.ThrottleStateUpdaterThread
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.plugin.periodical.Periodical annotated with @com.google.inject.internal.Element(setName=,uniqueId=378, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<org.graylog2.plugin.periodical.Periodical>
Sep 22 11:25:34 hostname graylog-server: for the 3rd parameter of org.graylog2.shared.initializers.PeriodicalsService.<init>(PeriodicalsService.java:45)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.initializers.PeriodicalsService.class(PeriodicalsService.java:36)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.initializers.PeriodicalsService
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.Service annotated with @com.google.inject.internal.Element(setName=,uniqueId=6, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<com.google.common.util.concurrent.Service>
Sep 22 11:25:34 hostname graylog-server: for field at org.graylog2.shared.bindings.providers.ServiceManagerProvider.services(ServiceManagerProvider.java:34)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.bindings.providers.ServiceManagerProvider
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:63)
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.ServiceManager
Sep 22 11:25:34 hostname graylog-server: Caused by: java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.buildFingerprint(StreamListFingerprint.java:55)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.<init>(StreamListFingerprint.java:37)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine.<init>(StreamRouterEngine.java:83)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine$$FastClassByGuice$$c6190f39.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.assistedinject.FactoryProvider2.invoke(FactoryProvider2.java:836)
Sep 22 11:25:34 hostname graylog-server: at com.sun.proxy.$Proxy43.create(Unknown Source)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.getNewEngine(StreamRouter.java:136)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.access$000(StreamRouter.java:102)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter.<init>(StreamRouter.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$$FastClassByGuice$$f9455617.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: Exception in thread "main" com.google.inject.CreationException: Unable to create injector, see the following errors:
Sep 22 11:25:34 hostname graylog-server: 1) Error injecting constructor, java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.buffers.ProcessBuffer.<init>(ProcessBuffer.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:59)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.buffers.ProcessBuffer
Sep 22 11:25:34 hostname graylog-server: for the 2nd parameter of org.graylog2.periodical.ThrottleStateUpdaterThread.<init>(ThrottleStateUpdaterThread.java:76)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.periodical.ThrottleStateUpdaterThread
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.plugin.periodical.Periodical annotated with @com.google.inject.internal.Element(setName=,uniqueId=378, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<org.graylog2.plugin.periodical.Periodical>
Sep 22 11:25:34 hostname graylog-server: for the 3rd parameter of org.graylog2.shared.initializers.PeriodicalsService.<init>(PeriodicalsService.java:45)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.initializers.PeriodicalsService.class(PeriodicalsService.java:36)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.initializers.PeriodicalsService
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.Service annotated with @com.google.inject.internal.Element(setName=,uniqueId=6, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<com.google.common.util.concurrent.Service>
Sep 22 11:25:34 hostname graylog-server: for field at org.graylog2.shared.bindings.providers.ServiceManagerProvider.services(ServiceManagerProvider.java:34)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.bindings.providers.ServiceManagerProvider
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:63)
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.ServiceManager
Sep 22 11:25:34 hostname graylog-server: Caused by: java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.buildFingerprint(StreamListFingerprint.java:55)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.<init>(StreamListFingerprint.java:37)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine.<init>(StreamRouterEngine.java:83)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine$$FastClassByGuice$$c6190f39.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.assistedinject.FactoryProvider2.invoke(FactoryProvider2.java:836)
Sep 22 11:25:34 hostname graylog-server: at com.sun.proxy.$Proxy43.create(Unknown Source)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.getNewEngine(StreamRouter.java:136)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.access$000(StreamRouter.java:102)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter.<init>(StreamRouter.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$$FastClassByGuice$$f9455617.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: Exception in thread "main" com.google.inject.CreationException: Unable to create injector, see the following errors:
Sep 22 11:25:34 hostname graylog-server: 1) Error injecting constructor, java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.buffers.ProcessBuffer.<init>(ProcessBuffer.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:59)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.buffers.ProcessBuffer
Sep 22 11:25:34 hostname graylog-server: for the 2nd parameter of org.graylog2.periodical.ThrottleStateUpdaterThread.<init>(ThrottleStateUpdaterThread.java:76)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.periodical.ThrottleStateUpdaterThread
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.plugin.periodical.Periodical annotated with @com.google.inject.internal.Element(setName=,uniqueId=378, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<org.graylog2.plugin.periodical.Periodical>
Sep 22 11:25:34 hostname graylog-server: for the 3rd parameter of org.graylog2.shared.initializers.PeriodicalsService.<init>(PeriodicalsService.java:45)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.initializers.PeriodicalsService.class(PeriodicalsService.java:36)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.initializers.PeriodicalsService
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.Service annotated with @com.google.inject.internal.Element(setName=,uniqueId=6, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<com.google.common.util.concurrent.Service>
Sep 22 11:25:34 hostname graylog-server: for field at org.graylog2.shared.bindings.providers.ServiceManagerProvider.services(ServiceManagerProvider.java:34)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.bindings.providers.ServiceManagerProvider
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:63)
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.ServiceManager
Sep 22 11:25:34 hostname graylog-server: Caused by: java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.buildFingerprint(StreamListFingerprint.java:55)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.<init>(StreamListFingerprint.java:37)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine.<init>(StreamRouterEngine.java:83)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine$$FastClassByGuice$$c6190f39.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.assistedinject.FactoryProvider2.invoke(FactoryProvider2.java:836)
Sep 22 11:25:34 hostname graylog-server: at com.sun.proxy.$Proxy43.create(Unknown Source)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.getNewEngine(StreamRouter.java:136)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.access$000(StreamRouter.java:102)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter.<init>(StreamRouter.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$$FastClassByGuice$$f9455617.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: Exception in thread "main" com.google.inject.CreationException: Unable to create injector, see the following errors:
Sep 22 11:25:34 hostname graylog-server: 1) Error injecting constructor, java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.buffers.ProcessBuffer.<init>(ProcessBuffer.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:59)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.buffers.ProcessBuffer
Sep 22 11:25:34 hostname graylog-server: for the 2nd parameter of org.graylog2.periodical.ThrottleStateUpdaterThread.<init>(ThrottleStateUpdaterThread.java:76)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.periodical.ThrottleStateUpdaterThread
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.plugin.periodical.Periodical annotated with @com.google.inject.internal.Element(setName=,uniqueId=378, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<org.graylog2.plugin.periodical.Periodical>
Sep 22 11:25:34 hostname graylog-server: for the 3rd parameter of org.graylog2.shared.initializers.PeriodicalsService.<init>(PeriodicalsService.java:45)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.initializers.PeriodicalsService.class(PeriodicalsService.java:36)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.initializers.PeriodicalsService
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.Service annotated with @com.google.inject.internal.Element(setName=,uniqueId=6, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<com.google.common.util.concurrent.Service>
Sep 22 11:25:34 hostname graylog-server: for field at org.graylog2.shared.bindings.providers.ServiceManagerProvider.services(ServiceManagerProvider.java:34)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.bindings.providers.ServiceManagerProvider
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:63)
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.ServiceManager
Sep 22 11:25:34 hostname graylog-server: Caused by: java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.buildFingerprint(StreamListFingerprint.java:55)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.<init>(StreamListFingerprint.java:37)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine.<init>(StreamRouterEngine.java:83)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine$$FastClassByGuice$$c6190f39.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.assistedinject.FactoryProvider2.invoke(FactoryProvider2.java:836)
Sep 22 11:25:34 hostname graylog-server: at com.sun.proxy.$Proxy43.create(Unknown Source)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.getNewEngine(StreamRouter.java:136)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.access$000(StreamRouter.java:102)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter.<init>(StreamRouter.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$$FastClassByGuice$$f9455617.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: Exception in thread "main" com.google.inject.CreationException: Unable to create injector, see the following errors:
Sep 22 11:25:34 hostname graylog-server: 1) Error injecting constructor, java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.buffers.ProcessBuffer.<init>(ProcessBuffer.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:59)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.buffers.ProcessBuffer
Sep 22 11:25:34 hostname graylog-server: for the 2nd parameter of org.graylog2.periodical.ThrottleStateUpdaterThread.<init>(ThrottleStateUpdaterThread.java:76)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.periodical.ThrottleStateUpdaterThread
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.plugin.periodical.Periodical annotated with @com.google.inject.internal.Element(setName=,uniqueId=378, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<org.graylog2.plugin.periodical.Periodical>
Sep 22 11:25:34 hostname graylog-server: for the 3rd parameter of org.graylog2.shared.initializers.PeriodicalsService.<init>(PeriodicalsService.java:45)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.initializers.PeriodicalsService.class(PeriodicalsService.java:36)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.initializers.PeriodicalsService
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.Service annotated with @com.google.inject.internal.Element(setName=,uniqueId=6, type=MULTIBINDER, keyType=)
Sep 22 11:25:34 hostname graylog-server: while locating java.util.Set<com.google.common.util.concurrent.Service>
Sep 22 11:25:34 hostname graylog-server: for field at org.graylog2.shared.bindings.providers.ServiceManagerProvider.services(ServiceManagerProvider.java:34)
Sep 22 11:25:34 hostname graylog-server: while locating org.graylog2.shared.bindings.providers.ServiceManagerProvider
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.shared.bindings.GenericBindings.configure(GenericBindings.java:63)
Sep 22 11:25:34 hostname graylog-server: while locating com.google.common.util.concurrent.ServiceManager
Sep 22 11:25:34 hostname graylog-server: Caused by: java.lang.NullPointerException
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.buildFingerprint(StreamListFingerprint.java:55)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamListFingerprint.<init>(StreamListFingerprint.java:37)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine.<init>(StreamRouterEngine.java:83)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouterEngine$$FastClassByGuice$$c6190f39.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.assistedinject.FactoryProvider2.invoke(FactoryProvider2.java:836)
Sep 22 11:25:34 hostname graylog-server: at com.sun.proxy.$Proxy43.create(Unknown Source)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.getNewEngine(StreamRouter.java:136)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$StreamRouterEngineUpdater.access$000(StreamRouter.java:102)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter.<init>(StreamRouter.java:62)
Sep 22 11:25:34 hostname graylog-server: at org.graylog2.streams.StreamRouter$$FastClassByGuice$$f9455617.newInstance(<generated>)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
Sep 22 11:25:34 hostname graylog-server: at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
```
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
Unkwown
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
I've create a brand new server and installed a 3.3 graylog server and connected to the old cluster version 3.2 but the error occurred again. I've tried before to upgrade the cluster, but had to rollback because unable to start the graylog instance.

If necessary, I can upload the mongo and ElasticSearch conf
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->


## Context
I can't upgrade graylog to the latest availble version
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

* Graylog Version: 3.3.5-1
* Java Version: 1.8.0_262
* Elasticsearch Version: elasticsearch-oss-6.8.12-1.
* MongoDB Version: godb-org-server-4.0.20-1
* Operating System: CentOS Linux release 7.8.2003 (Core)
* Browser version: N/A

--- a/graylog2-server/src/main/java/org/graylog2/streams/StreamServiceImpl.java
+++ b/graylog2-server/src/main/java/org/graylog2/streams/StreamServiceImpl.java
@@ -219,15 +219,14 @@ public class StreamServiceImpl extends PersistedServiceImpl implements StreamSer
             final Set<Output> outputs = outputIdsForRawStream(o)
                     .stream()
                     .map(ObjectId::toHexString)
-                    .map(outputsById::get)
-                    .collect(Collectors.toSet());
-
-            if (outputs.contains(null)) {
-                final String streamTitle = Strings.nullToEmpty((String)o.get(StreamImpl.FIELD_TITLE));
-                LOG.warn("Stream \"" + streamTitle + "\" <" + id + "> references missing outputs - ignoring these outputs.");
-            }
-
-            final Set<Output> nonNullOutputs = outputs.stream()
+                    .map(outputId -> {
+                        final Output output = outputsById.get(outputId);
+                        if (output == null) {
+                            final String streamTitle = Strings.nullToEmpty((String)o.get(StreamImpl.FIELD_TITLE));
+                            LOG.warn("Stream \"" + streamTitle + "\" <" + id + "> references missing output <" + outputId + "> - ignoring output.");
+                        }
+                        return output;
+                    })
                     .filter(Objects::nonNull)
                     .collect(Collectors.toSet());
 
@@ -236,7 +235,7 @@ public class StreamServiceImpl extends PersistedServiceImpl implements StreamSer
 
             final String indexSetId = (String)fields.get(StreamImpl.FIELD_INDEX_SET_ID);
 
-            streams.add(new StreamImpl(objectId, fields, streamRules, nonNullOutputs, indexSets.get(indexSetId)));
+            streams.add(new StreamImpl(objectId, fields, streamRules, outputs, indexSets.get(indexSetId)));
         }
 
         return streams.build();

--- a/graylog2-server/src/main/java/org/graylog2/streams/StreamServiceImpl.java
+++ b/graylog2-server/src/main/java/org/graylog2/streams/StreamServiceImpl.java
@@ -16,6 +16,7 @@
  */
 package org.graylog2.streams;
 
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Lists;
@@ -221,12 +222,21 @@ public class StreamServiceImpl extends PersistedServiceImpl implements StreamSer
                     .map(outputsById::get)
                     .collect(Collectors.toSet());
 
+            if (outputs.contains(null)) {
+                final String streamTitle = Strings.nullToEmpty((String)o.get(StreamImpl.FIELD_TITLE));
+                LOG.warn("Stream \"" + streamTitle + "\" <" + id + "> references missing outputs - ignoring these outputs.");
+            }
+
+            final Set<Output> nonNullOutputs = outputs.stream()
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toSet());
+
             @SuppressWarnings("unchecked")
             final Map<String, Object> fields = o.toMap();
 
             final String indexSetId = (String)fields.get(StreamImpl.FIELD_INDEX_SET_ID);
 
-            streams.add(new StreamImpl(objectId, fields, streamRules, outputs, indexSets.get(indexSetId)));
+            streams.add(new StreamImpl(objectId, fields, streamRules, nonNullOutputs, indexSets.get(indexSetId)));
         }
 
         return streams.build();

